### ISSUE TITLE
Intended audience/use?

### ISSUE THREAD

**Issue by cmaimone** [USRSE/find-an-rse#1]
Who is the intended audience for this?  For RSEs looking to connect with other RSEs?  As a member directory?  For people looking to hire an RSE?

**Comment by vsoch** [comment]
I don’t see why both wouldn’t work :)

**Comment by cmaimone** [comment]
If it's a member (or community) directory, it would be clearer to label it as such.  I wouldn't think to list myself in a Find an RSE page necessarily, but I would in a member directory.  If it's explicitly intended to connect people, then it would need to contain info that people can use to connect on -- location, technologies/languages, research fields, maybe what conferences they can be found at, etc.  And some folks may want to share that info with other US-RSE members but not publicly (which I realize we can do/work on, but it also requires thought about what membership is and some other bigger issues).  

If it's for people to hire an RSE, that's different.  At least some (maybe many) RSE aren't looking for positions or additional work.  How will external folks distinguish between them?  And do we want folks contacting individual RSEs unsolicited (I wouldn't want people pinging me about jobs, or with questions about coding, etc.)?  If it's to provide a way for RSEs to find positions, maybe we can include that specifically as a partner to the job listing site -- a Work with an RSE page.

**Comment by vsoch** [comment]
It's not really that specific, because it indexes the content of the entire page of the RSE (and he/she can write anything). An example use case is, let's say I'm a researcher at Stanford and I want to find help with Cromwell. If I were to do a search, I'd want to hit any RSEs at Stanford that work on it (by way of mention in their description) or any at the Broad Institute (if there is a community there). It could also be the case that the result hits a "Support" page that mentions Cromwell. For these reasons, it's not really like you would expect a directory. It's open text / free search over pages that are relevant to people, support, and services.

**Comment by vsoch** [comment]
> If it's explicitly intended to connect people, then it would need to contain info that people can use to connect on -- location, technologies/languages, research fields, maybe what conferences they can be found at, etc.

The RSE is free to write whatever they please on their profile page (I have expertise in these languages, I work on these projects, here are conferences I like to go to) so really the best thing is to have a single search box, akin to Google. Putting custom fields (and forcing people to fill them in) seems a bit like overfitting (and people won't want to do it).

**Comment by vsoch** [comment]
>  And some folks may want to share that info with other US-RSE members but not publicly (which I realize we can do/work on, but it also requires thought about what membership is and some other bigger issues).

This is a different use case - all of the site information is public, and not intended for private information. It is not technically possible to host private GitHub pages. But that's a good idea (for some other venue) -   maybe bring up in usrse slack about some (private) place to store things... what would it be used for?

**Comment by vsoch** [comment]
> If it's for people to hire an RSE, that's different. At least some (maybe many) RSE aren't looking for positions or additional work.

It's not. The main usrse site also has a [jobs page](https://us-rse.org/jobs/), so I imagine they would just go there :) The assumption is that the community pages from which the search is derived include individuals in (already employed) and established groups.

I hope I've answered your questions!

**Comment by cmaimone** [comment]
Your example case suggests that "Find an RSE" is to find RSEs for help:

> An example use case is, let's say I'm a researcher at Stanford and I want to find help with Cromwell. 

Suggesting that researchers can find an RSE for help gets at either 1) hiring, 2) service delivery, or 3) people asking for free help.  I'm hoping we agree that #3 isn't what we want to enable.  

For #1 and #2, I understand that many folks need to find their own projects and support for their positions, and this mixes service offerings with hiring.  

If some combo of #1 and #2 is the focus for the site, then the intended audience for the site is primarily people external to US-RSE membership.  This is fine, but I'd like to be clear in 1) the target users/audience, and 2) intended use cases.  Defining these things is important for understanding and evaluating developments of this site/functionality and for getting RSEs to actually use it.  

Part of what I'm looking for is the short explanation/blurb to US-RSE members about what this directory is and how we envision people (which people?) using is.  Otherwise US-RSE people won't know what to enter or if they want to be listed at all.  I think RSEs also need something of a value proposition as to whether adding their info and maintaining it is worth their time.

You also mention a "support" page  -- I'm not sure what this is referring to.

**Comment by vsoch** [comment]
> Suggesting that researchers can find an RSE for help gets at either 1) hiring, 2) service delivery, or 3) people asking for free help. I'm hoping we agree that #3 isn't what we want to enable.

I disagree, if I'm a student it has nothing to do with hiring, I want to find the people that are connected to an open source project that can answer my questions.  For larger things, I agree some official hiring or service is as bay. But for a user that wants to get help from services and people that already exist, it's a different scenario. A better example is perhaps a user that is sitting at Stanford needing help with running an analysis. Stanford has several research computing centers, service desks, and it might not be clear if he/she should reach out to general research computing, or maybe the SCG cluster? the library data services? or maybe one of the resources offered by Computer Science? When you realize the hairball that is IT, and you just have a simple question, it's overwhelming. Right now, getting help at Stanford is a hairball of submitting "Service Now" tickets (or asking around) and very commonly getting thrown around the routing system and having it take a few days to get help. What we need in this case is a portal where (all of these various groups) have written up what they offer that can be searched. And I also want to point out this portal is fundamentally different from something like service now - it's a home grown community page. The community can choose to direct users back to their service centers (e.g., a website, help forum, or similar) or direct them to post issues on the GitHub board. It's totally up to the community how they want to roll their space :)

As an RSE at Stanford I am paid of course to work on specific projects. But if a student or researcher comes to me and needs help? I am going to help them. No questions asked.  Even if there is a service opportunity (and it's not what you call "free help") it is my duty to respond to direct them to that. The point isn't to divy out services and resources at first contact, but to help people find them in the first place.

**Comment by vsoch** [comment]
> Part of what I'm looking for is the short explanation/blurb to US-RSE members about what this directory is and how we envision people (which people?) using is. Otherwise US-RSE people won't know what to enter or if they want to be listed at all. I think RSEs also need something of a value proposition as to whether adding their info and maintaining it is worth their time.

This is a fantastic idea - I've done a ton of work to make the search infrastructure, and I was (hoping) that people would step in to work on details like that, and bring them up as issues as you have. I can't speak for others, but for me there is value here because I *want* users working on software that might cross what I can help with to find me. 

> You also mention a "support" page -- I'm not sure what this is referring to.

Each community template has a support page where a community can write about centers, services, pretty much anything that they offer! For example, here is the one we are working on for Stanford -> https://us-rse.org/community-stanford/support/ and is suspect we will add a "Data" table to that :)

**Comment by vsoch** [comment]
And another huge deal that I haven't mentioned is establishing the RSE community, period. Right now funding for RSEs has to come from PIs, so many sit in labs. It hasn't been proven on the university level that this kind of SWE, or having a layer of them sitting at an institution, is important. There is no "career track" for an RSE. By establishing these small portals and then increasing awareness for RSEs, we can (slowly) move toward a future that doesn't look like the present.

**Comment by vsoch** [comment]
And other use case (for the RSE) is to find others that are working on the same projects! For example, I work remotely, and it gets a bit lonely. But let's say that I could search for the software that I work on, and then find others that are sitting in random departments at a university close by? Or if I travel, I could do the same? What we are doing is giving exposure to these individuals so we can better collaborate and find one another.

**Comment by cmaimone** [comment]
> I want to find the people that are connected to an open source project that can answer my questions.

This is should be facilitated through the open source project in question.

**Comment by vsoch** [comment]
Not necessarily, if someone at Stanford needs help with singularity containers, they should best come to me for detailed help. If they were to just go to the singularity community, it would be harder to get that help.

**Comment by cmaimone** [comment]
> A better example is perhaps a user that is sitting at Stanford needing help with running an analysis. Stanford has several research computing centers, service desks, and it might not be clear if he/she should reach out to general research computing, or maybe the SCG cluster? the library data services? or maybe one of the resources offered by Computer Science? When you realize the hairball that is IT, and you just have a simple question, it's overwhelming. Right now, getting help at Stanford is a hairball of submitting "Service Now" tickets (or asking around) and very commonly getting thrown around the routing system and having it take a few days to get help.

I agree, but this is an issue to address at the institutional (or group, or IT) level.  It is an issue in many places, but as a service manager, I can't support maintaining multiple portals and resource lists.  I don't see how fixing IT service management issues is within the scope of US-RSE activities.

**Comment by vsoch** [comment]
Then you don’t need to :-)  Participation is always voluntary for those that would benefit from exposing their resources, finding one another, etc.

**Comment by vsoch** [comment]
I can’t speak for all groups, but my manager doesn’t have time or resources to grow an RSE community. This way, I’m empowered to help, and to inspire change.

**Comment by vsoch** [comment]
And IT services (in my mind) is very different from Research Software Engineering. There are enormous numbers of help desks for that.

**Comment by cmaimone** [comment]
I agree with many of the things you're saying and challenges you're pointing out, but that does not necessarily mean all of those issues should be addressed on a platform that is associated with the US-RSE organization.  

Any platform or tool we have as part of the organization should have a clear intent/goal and audience.  Members should know how to use them.  And currently, as someone who would be involved in developing a community site, I don't have the context I need to participate/contribute.  I also don't have the info I need to write documentation or promote this for US-RSE members to use.  [I am someone who is voluntarily trying to use and promote what you've built!]

Providing a mechanism to link to (e.g. the map and organization list) or even search multiple sites is helpful, as is helping groups envision what their own community site might look like.  But I'm not sure that having lots of separate community sites with the US-RSE logo and organizational name is the right strategy (it implies local chapters or something similar to me).  Branding those sites as an official part of the org could result in problematic content, and it could also be confusing for researchers and non-members who aren't clear on the distinction between a community site and the main organization.

I think we need to have an actual conversation with a larger group to make progress on some of these issues. Perhaps we pick this up when we have some committees formed soon.

**Comment by vsoch** [comment]
Note that this is exactly what the UK RSE group did - https://rse.ac.uk/community/research-software-groups-rsgs/ -  the group pages are under the same branding and the only difference is perhaps the hosting is not modular,  and I’ve created a template that allows for a lot more rich content and customization. Having separate github repos with the same branding allows for the best ownership of the content and permissions for the group in question.

 I agree that it would be good to have discussion around this because we need to empower small groups to establish themselves in the community, and of course have clearly stated goals and organization.
